presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoResultSet.java

```diff
@@ -150,7 +153,8 @@ public String getQueryId()
 
     public QueryStats getStats()
     {
-        return QueryStats.create(client.getStats());
+        QueryResults current = client.isValid() ? client.current() : client.finalResults();
```

Comment:
`current()`, `finalResults()` and `getStats()` do the same thing, `currentResults.get()`. The only difference is the `checkState()` calls, which `getStats()` does not do at all.